### PR TITLE
Fix SSL host name

### LINKED ISSUE #433: Python client adds unexpected `riak@` to certificate CN check [JIRA: CLIENTS-753]
In https://github.com/basho/riak-python-client/blame/master/riak/transports/pbc/connection.py#L133 the prefix `riak@` is added to the SSL certificate host check. Therefore the CN or SAN records of the server certificate need to be `riak@<hostname>` or riak@<ip_addr>`, whereas it's common practice to issue server certificate for the host-name or ip address of the server. The client behaviour is therefore unexpected. Furthermore, this behaviour is unique to the python and ruby clients and generally completely undocumented in Riak's server documentation.

The test certificate has a CN record of `riak@127.0.0.1`, which should be changed accordingly.

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #433 (CLIENTS-753) and #434 (CLIENTS-765)

Depends on basho/riak-client-tools#4

### CODE REVIEW

File: riak/tests/test_security.py [hunk 1]
```diff
@@ -91,7 +90,6 @@ def test_security_password_without_cacert(self):
     @unittest.skipUnless(RUN_SECURITY, 'RUN_SECURITY is 0')
     def test_security_cert_authentication(self):
         creds = SecurityCreds(username=SECURITY_CERT_USER,
-                              password=SECURITY_CERT_PASSWD,
```
Comment (lukebakken) [inline riak/tests/test_security.py]: NB: password is *not* required for cert-based auth.

File: riak/transports/pbc/connection.py [hunk 2]
```diff
@@ -84,7 +84,10 @@ def _auth(self):
         """
         req = riak.pb.riak_pb2.RpbAuthReq()
         req.user = str_to_bytes(self._client._credentials.username)
-        req.password = str_to_bytes(self._client._credentials.password)
+        password = self._client._credentials.password
+        if not password:
+            password = ''
+        req.password = str_to_bytes(password)
```
Comment (lukebakken) [inline riak/transports/pbc/connection.py]: Even though password is not required by cert-based auth, it is required in the PB message, which is why an empty string is encoded here.

File: MANIFEST.in [hunk 3]
```diff
@@ -1,6 +1,6 @@
 include docs/*
 include riak/erl_src/*
-include README.rst
+include README.md
```
Comment (javajolt) [inline MANIFEST.in]: Nice to be consistent. It's all MarkDown these days

**Comment by javajolt** [comment]
Now that I tweaked my environment, life it good

**Comment by javajolt** [comment]
:+1: 550a2d1

**Comment by lukebakken** [comment]
@borshop merge

**Pull request merged**